chef/automate
# Removing inner transaction

### :nut_and_bolt: Description: What code changed, and why?
What we are trying to solve with this PR is removing the piece of code that creates a transaction inside of a transaction. 
When the outer transaction failed a new one was created inside the existing one. The solution was to create a postgresql function that first tries to insert the message with the ID. If there was no row with the ID, insert the row with the cloud information. If there is already cloud information matching, that matching row is updated.

### :chains: Related Resources
#4172 

### :athletic_shoe: How to Build and Test the Change
There is no change, but to exercise the changed code:
* `build components/nodemanager-service && start_all_services`
* Update the converge-success-report.json file. Add two fields to the "ec2" section.  https://github.com/chef/automate/blob/242446ba743f8cb385f53cddd9a53f2b7b45b8c3/components/ingest-service/examples/converge-success-report.json#L1460
```
        "account_id": "999999999999",
        "region": "eu-west-1",
```
* Ingest the a CCR with `cat /src/components/ingest-service/examples/converge-success-report.json | send_chef_data_raw`
* Ensure the three cloud information fields are set in the database. Use `chef-automate dev psql nodemanager_service`. Then `select id, source_region, source_account_id, source_id from nodes;`
* Then update the one or all the cloud information fields (account_id, region, instance_id) in the converge-success-report.json file
* Ingest another CCR with the same ID as before by running this command `cat /src/components/ingest-service/examples/converge-success-report.json | send_chef_data_raw`
* Ensure whatever changes were made to the cloud information fields are updated in the database with the same steps from above. 

### :white_check_mark: Checklist
